app/models/user.rb

```diff
@@ -27,6 +28,10 @@ def sysadmin=(value)
     end
   end
 
+  def find_role(role, admin_set)
```

Comment:
This looks problematic and unused. You'd need resource_type to find a unique role.